### ISSUE TITLE
Overwriting rule options does not work

### ISSUE THREAD

**Issue by CSchulz** [palantir/tslint#3059]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
console.log()
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint-microsoft-contrib",
  "rules": {
    "no-console": {
      "options": [
        "time",
        "timeEnd"
      ]
    }
  }
}
```

tslint-microsoft-contrib:
```json
    "no-console": [
      true,
      "debug",
      "info",
      "log",
      "time",
      "timeEnd",
      "trace"
    ],
```

#### Actual behavior
Does not fail for console.log.

#### Expected behavior
Should fail.

**Comment by ajafff** [comment]
The options for this rule are a blacklist, not a whitelist.
If you don't specify any banned methods, all are banned by default. If you specify options, only those methods are banned.

We should definitely add the ability to whitelist certain methods instead of blacklisting all other methods. But that's out of scope for this issue.

**Comment by CSchulz** [comment]
I would like to implement it as whitelist, but I don't see any good way to implement it without breaking changes.

Do you have any hint for me?

**Comment by ajafff** [comment]
@CSchulz #924 already contains some thoughts on this topic. This discussion should continue over there.
